### ISSUE TITLE
Limiting Train Size per epoch for large dataset

### ISSUE THREAD

**Issue by edwardpwtsoi** [PyTorchLightning/pytorch-lightning#7364]
## 🚀 Feature
Limit train batch work on non debugging setting.

### Motivation
#5725 suffering from the same problem, I think limiting the train batch size could help. But I am not sure how the behaviour after setting the limit train batch. Would it train on the same portion of data every epoch?

### Pitch
This would be a nice work around and if it could guarantee it would step through all datasets, it would be wonderful.

### Additional context

I was working on a datasets of size nearly 0.5TB

**Comment by tchaton** [comment]
Dear @edwardpwtsoi,

`limit_train_batches` should return different batches after each epoch as the dataset is being shuffled.

Best
T.C
